TFG-Labs/custom-shipping-step-by-items
# [WEB-476] Bpf3309 - Fix Bad Addresses created by mobile

### What problem is this solving?
Geo-Cordinates create by a previous mobile app can come in the following form: 
` "geoCoordinate": "[28.304687,-25.7020923]",`

In the following three scenarios we do not want to lose these geocordinates on the orderform or in master data
1. When selecting an address
2. On Clicking proceed to payment
3. On Edit 


#### How it works
- we also fixed an issue with the base url for the api -> the previous implementation always returned the staging api for thefoschini workspaces

- on reading in addresses to load into index db we 'heal the co ordinate data'

#### How to test it?
Look at your addresses API request and see an address in that format
<img width="835" alt="image" src="https://github.com/user-attachments/assets/ce158bf1-e397-4872-8fe9-8dc557e85d5d">


1. When selecting an address
Select that address 
visit your orderform
https://bpf3309--thefoschini.myvtex.com/api/checkout/pub/orderForm/{ofid}

see the co cordinates correctly mapped:
<img width="599" alt="image" src="https://github.com/user-attachments/assets/2783f7f1-ee10-40f1-84fe-7ca2f408f127">

2. On Clicking proceed to payment
<img width="599" alt="image" src="https://github.com/user-attachments/assets/638035a9-3e11-415f-b58a-a73432514098">

3. On Edit 
<img width="599" alt="image" src="https://github.com/user-attachments/assets/d5978f8a-dc23-4a91-992c-fc897508c175">

https://bpf3309--thefoschini.myvtex.com/checkout/#/shipping


### Related to / Depends on
None 

#### How does this PR make you feel? :link:
good
